client/src/partials/journal/modals/references.modal.html

```diff
@@ -0,0 +1,96 @@
+<ng-form name="ModalForm">
+  <div class="modal-header">
+    <h4>{{ "MODALS.REFERENCES.TITLE" | translate }}</h4>
+  </div>
+
+  <!-- main form body -->
+  <div class="modal-body">
+
+    <!-- dropdown to choose receipt type -->
+    <div class="form-group" ng-class="{ 'has-error' : ModalForm.$submitted && ModalForm.type.$error }">
+      <label class="control-label">{{ "MODALS.REFERENCES.TYPE" | translate }}</label>
+      <select
+        class="form-control"
+        name="type"
+        ng-model="ReferenceLookupModalCtrl.target"
+        ng-options="tgt as (tgt.key | translate) disable when tgt.disabled for tgt in ::ReferenceLookupModalCtrl.targets | orderBy:tgt.key"
+        required
+        >
+        <option value="" disabled>{{ "SELECT.RECEIPT" | translate }}</option>
+      </select>
+    </div>
+
+    <!-- input to find the correct receipt by reference -->
+    <div class="form-group has-feedback" ng-class="{
+      'has-error' : ModalForm.reference.$dirty && ModalForm.reference.$invalid
+    }">
+      <label class="control-label">{{ "MODALS.REFERENCES.REFERENCE_LOOKUP" | translate }}</label>
+      <div class="input-group">
+        <input
+          type="text"
+          name="reference"
+          class="form-control"
+          ng-model="ReferenceLookupModalCtrl.reference"
+          ng-disabled="(ReferenceLookupModalCtrl.loading || !ReferenceLookupModalCtrl.target)"
+          required>
+        <span class="input-group-btn">
+          <button
+            class="btn btn-default"
+            type="button"
+            ng-click="ReferenceLookupModalCtrl.lookupReference()"
+            ng-disabled="(ReferenceLookupModalCtrl.loading || !ReferenceLookupModalCtrl.target)">
+            <span ng-hide="vm.loading"><span class="glyphicon glyphicon-search"></span> Search</span>
+            <span ng-show="vm.loading"><span class="glyphicon glyphicon-time"></span> Loading</span>
+          </button>
+        </span>
+      </div>
+
+      <div class="help-block" ng-messages="ModalForm.reference.$error" ng-show="ModalForm.reference.$dirty">
+        <div ng-messages-include="partials/templates/messages.tmpl.html"></div>
+      </div>
+
+      <!-- success block -->
+      <p class="help-block" ng-show="ModalForm.reference.$dirty && ReferenceLookupModalCtrl.receipt">
+        <span class="text-success">
+          <span class="glyphicon glyphicon-ok"></span> Successfully found a receipt!
```

Comment:
There are a few constant strings here in the UI - these should be translation keys.